packages/@divvi/mobile/src/send/SendConfirmation.test.tsx

```diff
@@ -159,31 +166,43 @@ describe('SendConfirmation', () => {
     expect(getByTestId('ConfirmButton')).toHaveTextContent('send')
   })
 
-  it('prepares a transaction on load', () => {
-    renderScreen()
-    expect(mockUsePrepareSendTransactionsOutput.clearPreparedTransactions).toHaveBeenCalledWith()
-    jest.advanceTimersByTime(300)
-    expect(mockUsePrepareSendTransactionsOutput.refreshPreparedTransactions).toHaveBeenCalledTimes(
-      1
-    )
-    expect(mockUsePrepareSendTransactionsOutput.refreshPreparedTransactions).toHaveBeenCalledWith({
-      amount: mockTokenTransactionData.tokenAmount,
-      token: mockCusdTokenBalance,
-      recipientAddress: mockTokenTransactionData.recipient.address,
-      walletAddress: mockAccount.toLowerCase(),
-      feeCurrencies: mockFeeCurrencies,
-    })
+  it('does not prepare a transaction when navigated from SendEnterAmount', () => {
+    renderScreen(mockSendConfirmationProps)
+    expect(mockUsePrepareSendTransactionsOutput.clearPreparedTransactions).not.toHaveBeenCalled()
+    expect(mockUsePrepareSendTransactionsOutput.refreshPreparedTransactions).not.toHaveBeenCalled()
+    expect(mockUsePrepareSendTransactionsOutput.prepareTransactionLoading).toBe(false)
+    expect(mockUsePrepareSendTransactionsOutput.prepareTransactionError).toBeUndefined()
   })
 
-  it('disables send if transaction is not prepared yet', () => {
-    mockUsePrepareSendTransactionsOutput.prepareTransactionsResult = undefined
-    const { getByTestId } = renderScreen()
+  describe('when opened with a deeplink', () => {
+    it('prepares a transaction on load if the screen is opened with a deeplink', () => {
+      renderScreen(mockSendConfirmationFromExternalProps)
+      expect(mockUsePrepareSendTransactionsOutput.clearPreparedTransactions).toHaveBeenCalledWith()
```

Comment:
```suggestion
      
await waitFor(() =>
        expect(
          mockUsePrepareSendTransactionsOutput.refreshPreparedTransactions
        ).toHaveBeenCalledTimes(1)
      )      expect(mockUsePrepareSendTransactionsOutput.clearPreparedTransactions).toHaveBeenCalledWith()
```

using `waitFor` for the first assertion that confirms the component has reached a steady state is good practice. this ensures that subsequent assertions are made on the stable state rather than a transient one. currently, the assertion for `clearPreparedTransactions` runs while there are unresolved promises, and advancing fake timers by an arbitrary amount resolves them but using `waitFor` would allow us to not rely on the arbitrary number